src/transformers/trainer_utils.py

```diff
@@ -82,19 +82,24 @@ def enable_full_determinism(seed: int, warn_only: bool = False):
         tf.config.experimental.enable_op_determinism()
 
 
-def set_seed(seed: int):
+def set_seed(seed: int, determinsitic: bool = False):
     """
     Helper function for reproducible behavior to set the seed in `random`, `numpy`, `torch` and/or `tf` (if installed).
 
     Args:
-        seed (`int`): The seed to set.
+        seed (`int`):
+            The seed to set.
+        determinsitic (`bool`, *optional*, defaults to `False`):
+            Whether to use deterministic algorithms where available.
```

Comment:
As discussed, maybe mention enabling this might potentially slow down trainings.